code/game/objects/items/control_wand.dm

```diff
@@ -29,6 +29,28 @@
 	to_chat(user, "Now in mode: [mode].")
 
 /obj/item/weapon/door_remote/afterattack(obj/machinery/door/airlock/D, mob/user)
+	var/turf/doorlocation = get_turf(D)
+	if(istype(D)) //yes, the istype has to be here, otherwise it runtimes if you misclick
+		if(D.is_special)
+			if(D.check_access(src.ID))
+				if(istype(D, /obj/machinery/door/airlock/hatch/gamma) && !(seclevel2num(get_security_level()) == SEC_LEVEL_GAMMA))
```

Comment:
Err, all these type checks are worse than it was before. Just using a single generic message such as what you have at the bottom or `"[D] cannot be opened using [src]!"` would be fine.